internal/encryption/method/aesgcm/README.md

```diff
@@ -0,0 +1,54 @@
+# AES-GCM encryption method
+
+This folder contains the state encryption implementation of the AES-GCM encryption method.
+
+This is implemented following the guidance of the following document: ([NIST SP 800-38D](https://csrc.nist.gov/pubs/sp/800/38/d/final)).
+
+
+## Configuration
+
+You can configure the encryption by specifying the following method block:
+
+```hcl2
+terraform {
+  encryption {
+    method "aes_gcm" "mymethod" {
+      key        = "key here" # Pass your 16, 24, or 32 byte encryption key here.
+      aad        = ""         # Leave empty unless needed
+      nonce_size = 12         # Minimum: 1, do not change unless needed

```

Comment:
Given the changes to config, I think this can be removed?

Reply:
Yes, I still need to work on the docs, thank you!